src/Domain/CoordinateMaps/UniformCylindricalFlatEndcap.cpp

```diff
@@ -0,0 +1,712 @@
+      (center_two[2] - center_one[2] - r_one_cos_theta_one);
+  return square(target_coords[0] - center_one[0] -
+                lambda * (center_two[0] - center_one[0])) +
+         square(target_coords[1] - center_one[1] -
+                lambda * (center_two[1] - center_one[1])) -
+         square((1.0 - lambda) * radius_one * sin(rhobar * theta_max_one) +
+                lambda * radius_two * rhobar);
+}
+
+// min and max values of rhobar in the inverse function.
+std::array<double, 2> rhobar_min_max(
+    const std::array<double, 3>& center_one, const double radius_one,
+    const double theta_max_one, const std::array<double, 3>& target_coords) {
+  // Choose the minimum value of rhobar so that lambda >=0, where
+  // lambda is the quantity inside function_to_zero.  Note that the
+  // denominator of lambda inside that function is always positive.
+  const double rhobar_min =
+      target_coords[2] - center_one[2] >= radius_one
+          ? 0.0
+          : acos((target_coords[2] - center_one[2]) / radius_one) /
+                theta_max_one;
+  return {rhobar_min, 1.0};
+}
+
+// Returns whether the map is invertible if the target is on sphere_one.
+bool is_invertible_for_target_on_sphere_one(
+    const std::array<double, 3>& center_one,
+    const std::array<double, 3>& center_two, const double radius_one,
+    const double radius_two, const double theta_max_one,
+    const std::array<double, 3>& target_coords) {
+  const auto [rhobar_min, rhobar_max] =
+      rhobar_min_max(center_one, radius_one, theta_max_one, target_coords);
+  if (equal_within_roundoff(rhobar_min, rhobar_max)) {
+    // Special case where rhobar_min == rhobar_max to roundoff.  This
+    // case occurs when the target point (assumed to be on sphere_one)
+    // is at rhobar=1.  In this case there is nothing to test because
+    // there is only one possible value of rhobar, so there cannot be
+    // two roots for two different values of rhobar.
+    return true;
+  }
+  const size_t num_pts = 1000;
+  const double drhobar = (rhobar_max - rhobar_min) / (num_pts - 1.0);
+  // Because the target point is on sphere_one, the only root should
+  // be at rhobar = rhobar_min, where lambda=0.  The actual value of
+  // the function at rhobar_min may be either sign because of
+  // roundoff.  So ignore that point, and start at the next point.
+  // Keep track of the sign, and look for sign changes as we progress
+  // from point to point.
+  std::optional<double> sign;
+  for (size_t i = 1; i < num_pts; ++i) {
+    const double rhobar = rhobar_min + i * drhobar;
+    const double func = this_function_is_zero_for_correct_rhobar(
+        rhobar, center_one, center_two, radius_one, radius_two, theta_max_one,
+        target_coords);
+    if (not sign.has_value()) {
+      sign = func > 0.0 ? 1.0 : -1.0;
+    } else if (func * sign.value() <= 0.0) {
+      // There is another root, so function is not invertible.
+      return false;
```

Comment:
Do you understand why codecov is mad about this? Would it be hard to modify the unit test to handle this case? Same comment for all codecov failures